de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/SarosSessionManager.java

```diff
@@ -135,6 +146,13 @@ public void negotiationTerminated(final SessionNegotiation negotiation) {
         @Override
         public void negotiationTerminated(final ProjectNegotiation negotiation) {
             currentProjectNegotiations.remove(negotiation);
+
+            if (!currentProjectNegotiations.negotiationsInProgress()) {
+                /** notify about the completion of all running negotiations **/
+                synchronized (nextSharedResourcesWorkerNotify) {
+                    nextSharedResourcesWorkerNotify.notify();
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: "notify" may not wake up the appropriate thread. ([squid:S2446](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS2446))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=272&issues=AWaT0XAZyONdBoB89fGw&open=AWaT0XAZyONdBoB89fGw)

[ISSUE_KEY]: AWaT0XAZyONdBoB89fGw